### DISCUSSION TITLE
Exception Handling: Catch 400 (Bad Request) signInWithEmailAndPassword

### DISCUSSION THREAD

**Discussion by dharnil** [firebase/firebase-js-sdk#8082] [Q&A]
Hi Fellow Firebaser,

I am currently trying to validate incorrect credentials using `signInWithEmailAndPassword`. I see the following printed in my console why I initialise this function.

> Post **Endpoint + Key= {APIKEY} 400 (Bad Request)
![image](https://github.com/firebase/firebase-js-sdk/assets/116938206/321b0222-54b6-4815-bf5b-4829bb2b0fad)

My goal is to catch this message and prevent it from printing in console.

```
signInWithEmailAndPassword(auth, email, password)
        .then(res => console.log(res))
        .catch(error => console.log(error.code.replace('auth/', '')))
```

Also, can malicious actors hit the endpoint enough using your APIKEY that a firebase app reaches its request quota thus jeopardising users from being able to login?

I have tried to read the details on Firebase but there is no mention about this.

**Comment by dibakash** [comment]
I am using a bandaid solution by a quick and dirty console.clear() 
There has to be a better way to handle this.
